gui/websocket.go

```diff
@@ -2,11 +2,17 @@ package gui
 
 import (
 	"net/http"
+	"sync"
 
 	"github.com/gorilla/websocket"
 )
 
-var clients = make(map[*websocket.Conn]bool)
+type clientList struct {
```

Comment:
Why are you defining a new type instead of just adding a mutex (`var mtx sync.Mutex`) here?

Reply:
changed